hypothesis/lms
# Keep the LMSSegment table up to date in the python code

For: 

- https://github.com/hypothesis/lms/issues/6851


### Testing

- Trigger fetching group sets on this canvas course, for example re-deep-linking an assignment:

https://hypothesis.instructure.com/courses/125/assignments/1833/edit?return_to=https%3A%2F%2Fhypothesis.instructure.com%2Fcourses%2F125%2Fassignments%2F1833


No need to save the edit, just get to list the group sets.


- Launch both a sections and groups assignments:


https://hypothesis.instructure.com/courses/125/assignments/1833
https://hypothesis.instructure.com/courses/125/assignments/874


- Check the new rows in the DB:

```
 select * from lms_segment;
 id  |      type      | lms_id |              name               |         h_authority_provided_id          | lms_course_id | lms_group_set_id |          created           |          updated
-----+----------------+--------+---------------------------------+------------------------------------------+---------------+------------------+----------------------------+----------------------------
 117 | canvas_group   | 131    | Another test group              | e8c66a347a5f547ae87928d4e92e5045908f8d54 |           105 |              123 | 2024-11-12 09:58:47.927725 | 2024-11-12 09:58:47.927725
 118 | canvas_group   | 610    | Group created after course copy | 28b6d0a2ce206db35ef53624cb9fc66d5835fadd |           105 |              123 | 2024-11-12 09:58:47.927725 | 2024-11-12 09:58:47.927725
 119 | canvas_group   | 130    | Group with student              | 102a25279bc609179f58cb4bf5c50d7dfc6f0929 |           105 |              123 | 2024-11-12 09:58:47.927725 | 2024-11-12 09:58:47.927725
 120 | canvas_group   | 132    | Test group                      | 381c455de84ae2f6d2fc84142f4e898b29d748f3 |           105 |              123 | 2024-11-12 09:58:47.927725 | 2024-11-12 09:58:47.927725
 121 | canvas_section | 144    | Section 1                       | d2871e5e1c36b58f207d9ef5b22dcb7829f52e61 |           105 |                  | 2024-11-12 09:58:52.663572 | 2024-11-12 09:58:52.663572
 122 | canvas_section | 145    | Section 2                       | f453c6124e4ead4471896f26e896c3e5bbf0bb36 |           105 |                  | 2024-11-12 09:58:52.663572 | 2024-11-12 09:58:52.663572
 123 | canvas_section | 146    | Section 3                       | 83e64baad215f63057387910a1b41debc716a150 |           105 |                  | 2024-11-12 09:58:52.663572 | 2024-11-12 09:58:52.663572
```
